src/mvs/fusion.cc

```diff
@@ -246,27 +277,30 @@ void StereoFusion::Run() {
     Timer timer;
     timer.Start();
 
-    std::cout << StringPrintf("Fusing image [%d/%d]", num_fused_images + 1,
-                              model.images.size())
+    std::cout << StringPrintf("Fusing image [%d/%d] with index %d",
+                              num_fused_images + 1, model.images.size(),
+                              image_idx)
               << std::flush;
 
     const int width = depth_map_sizes_.at(image_idx).first;
     const int height = depth_map_sizes_.at(image_idx).second;
     const auto& fused_pixel_mask = fused_pixel_masks_.at(image_idx);
 
-    FusionData data;
-    data.image_idx = image_idx;
-    data.traversal_depth = 0;
-
-    for (data.row = 0; data.row < height; ++data.row) {
-      for (data.col = 0; data.col < width; ++data.col) {
-        if (fused_pixel_mask.Get(data.row, data.col)) {
+#pragma omp parallel for schedule(dynamic) num_threads(num_threads)
```

Comment:
Any OpenMP code needs to be wrapped in `#ifdef OPENMP_ENABLED`, because it's an optional dependency in Python. Alternatively and preferably, we could use the ThreadPool in `util/threading.h`. We could implement a

Reply:
Got it; I'll wrap it as such, and use your suggestion for an attempt to use the existing ThreadPool.

Reply:
Quick question on wrapping with `#ifdef OPENMP_ENABLED`. I thought that the `#pragma`s are ignored in cases where they cannot be resolved (i.e. OpenMP missing and finding a `#pragma omp` directive). Is this not the case with the colmap setup?

Reply:
It will certainly trigger an unknown pragma warning. If you then also treat warnings as errors, compilation will fail.